spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala

```diff
@@ -132,8 +143,187 @@ class TensorSample[T: ClassTag](
   }
 }
 
+/**
```

Comment:
In my testing, use a data array[T] and a size array[Int], we can save about 60 bytes for each sample.